modules/signal/src/signal_resample.cpp

```diff
@@ -0,0 +1,372 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#include "precomp.hpp"
+
+#include <opencv2/signal/signal_resample.hpp>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/core/hal/intrin.hpp>
+#include <opencv2/core/utils/trace.hpp>
+
+#include <algorithm>
+#include <cmath>
+
+namespace cv {
+namespace signal {
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+#define v_float32_width (uint32_t)VTraits<v_float32>::vlanes()
+const uint32_t v_float32_max_width = (uint32_t)VTraits<v_float32>::max_nlanes;
+#endif
+
+// Modified Bessel function 1st kind 0th order
+static float Bessel(float x)
+{
+    int k = 12; // approximation parameter
+    float defmul = powf(x, 2) * 0.25f;
+    float mul = defmul;
+    float acc = 0.f;
+    for(int i = 0 ; i < k; ++i)
+    {
+        mul = powf(defmul, static_cast<float>(i));
+        mul = mul / powf(tgammaf(static_cast<float>(i + 1)), 2.f); // tgamma(i+1) equals i!
+        acc +=mul;
+    }
+    return acc;
+}
+
+static void init_filter(float beta, int ntabs, float* tabs)
+{
+    auto fc = 0.25f;
+    // build sinc filter
+    for (int i = 0; i < ntabs; ++i)
+    {
+        tabs[i] = 2 * fc * (i - (ntabs - 1) / 2);
+    }
+    float *tmparr = new float[ntabs];
+    for (int i = 0 ; i < ntabs; ++i)
+    {
+        if (tabs[i] == 0.f)
+        {
+            tmparr[i] = 1.f;
+            continue;
+        }
+        tmparr[i] = (float)(CV_PI * tabs[i]);
+    }
+    // multiply by Kaiser window
+    for (int i = 0; i < ntabs; ++i)
+    {
+        tabs[i] = std::sin(tmparr[i]) / tmparr[i];
+        tabs[i] *= Bessel(beta * sqrtf((float)1 - powf((2 * i / (float)(ntabs - 1) - 1), 2))) / Bessel(beta);
+    }
+    float sum{0.f};
+    for (int i = 0 ; i < ntabs; ++i)
+    {
+        sum += tabs[i];
+    }
+    sum = 1.f/sum;
+    // normalize tabs to get unity gain
+    for (int i = 0; i < ntabs; ++i)
+    {
+        tabs[i] *= sum;
+    }
+    delete[] tmparr;
+}
+
+/////////////// cubic Hermite spline (tail of execIntrinLoop or scalar version) ///////////////
+static float scal_cubicHermite(float A, float B, float C, float D, float t)
+{
+    float a = (-A + (3.0f * B) - (3.0f * C) + D) * 0.5f;
+    float b = A + C + C - (5.0f * B + D) * 0.5f;
+    float c = (-A + C) * 0.5f;
+    return a * t * t * t + b * t * t + c * t + B;
+}
+
+/////////////// cubic Hermite spline (OpenCV's Universal Intrinsics) ///////////////
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+static v_float32 simd_cubicHermite(v_float32 v_A, v_float32 v_B, v_float32 v_C, v_float32 v_D, v_float32 v_t)
```

Comment:
> v_float32

1. `static inline`
2. `const v_float32&` to workaround alignment issues (see Win32 builder)